### DISCUSSION TITLE
normal chanel texture transformation is ignored

### DISCUSSION THREAD

**Discussion by orenmiron** [google/model-viewer#3278] [Q&A]
hi
my gltf contains a texture transformation for the normals chanel.
the transformation is ignored by model viewer
in this example, my normals map contains a white polygon.
see snapshots from  from khronos viewer (correct) and model viewer (wrong)
normals polygon is marked with red
after all snapshots, I attached the gltf inside a zip file.
please advise
![normals1](https://user-images.githubusercontent.com/7829843/158692300-1b0b9837-c26b-4408-a4c1-cf87df36bd7e.jpg)
![normals2](https://user-images.githubusercontent.com/7829843/158692317-aec91da5-5f13-4d6a-a074-3f945310b97a.jpg)
![normals_map](https://user-images.githubusercontent.com/7829843/158692350-dd383a80-454f-4b82-8a92-ea6353f1ce86.png)
[Rectangle.zip](https://github.com/google/model-viewer/files/8274199/Rectangle.zip)

**Comment by elalish** [comment]
You glTF is invalid, please see the errors reported in our [Editor](https://modelviewer.dev/editor/). Let us know if it still has problems once you export a version without errors.

**Reply by orenmiron** [reply]
Hi
Attaching a new gltf file without errors.
same question as before: why normal texture doesn't respect transformation.
I also downloaded the master branch and built it - same error
if you can direct me to the source file that load the gltf and where the normal transformation is used, I will try to fix it.
Thanks
[Rectangle.zip](https://github.com/google/model-viewer/files/8317275/Rectangle.zip)
.

**Reply by elalish** [reply]
This is below the level of our library; you'll need to look at Three.js and their GLTFLoader.js file. Can you repro in their editor? https://threejs.org/editor/

**Reply by orenmiron** [reply]
thanks.
it's a three.js problem.
can you recommend a discussion group where i can get support?
Is Three.js sources are on git hub?

**Reply by elalish** [reply]
You can file an issue here, and also a PR to fix it: https://github.com/mrdoob/three.js/issues
